app/job/rest.py

```diff
@@ -53,6 +55,15 @@ def get_job_by_service_and_job_id(service_id, job_id):
     return jsonify(data=data)
 
 
+@job.route('/<job_id>', methods=['DELETE'])
```

Comment:
is DELETE the semantically appropriate method? The job's still GETable afterwards. I would have leant towards

`POST /service/{}/job/{}/cancel`